### PR TITLE
Updated readme and added demographics table

### PR DESCRIPTION
**Pull request by achasmita**

### CODE REVIEW

File: docker-compose-dev.yml
Comment (shankari) [inline docker-compose-dev.yml:16]: This should not be hardcoded since the DB will change depending on the dataset used for testing. Please revert.

File: utils/constants.py
Comment (shankari) [inline utils/constants.py:92]: We cannot hardcode these because the survey can be different for each partner.
You should exclude the entries that the config wants to be excluded (currently marked as TODO) and potentially exclude several of the entries here. But we should not have a list of valid cols

File: utils/db_utils.py
Comment (shankari) [inline utils/db_utils.py:109]: demographic data is not the same as trip data - it is typically only answered once, at the beginning of the program. 
Trips come in throughout the program. We cannot blindly apply the trip filters to the demographic data

File: utils/permissions.py
Comment (shankari) [inline utils/permissions.py:92]: we should remove this (see note above)

File: pages/data.py
Comment (shankari) [inline pages/data.py:64]: is this still TODO?

File: app_sidebar_collapsible.py
Comment (shankari) [inline app_sidebar_collapsible.py:211]: note comment above: we don't need this

File: docker/start.sh
Comment (shankari) [inline docker/start.sh:10]: This is specific to the DB format that you are testing against and should not be checked in.

**Review by shankari — CHANGES_REQUESTED** [review]

File: utils/db_utils.py
Comment (shankari) [inline utils/db_utils.py:111]: I am not going to hold this up but have logged a potential improvement for later.

File: docker-compose-dev.yml
Comment (shankari) [inline docker-compose-dev.yml:50]: extraneous whitespace, revert. You could just revert the entire file

File: pages/data.py
Comment (shankari) [inline pages/data.py:65]: why are we not supporting excluded columns here? Please see my previous comment
https://github.com/e-mission/op-admin-dashboard/pull/63#discussion_r1313033861

>  You should exclude the entries that the config wants to be excluded (currently marked as TODO) and potentially exclude several of the entries here. But we should not have a list of valid cols

I don't see where you are excluding the entries that are marked as needing exclusion in the config.

**Review by shankari — CHANGES_REQUESTED** [review]
Almost there!

File: pages/data.py
Comment (shankari) [reply pages/data.py:65]: For backwards compatibility, if the config doesn't have the `data_demographic_columns_exclude`, then you can set it to a blank array. add a note to remove the backwards compat change after a couple of years when all the current studies are done.

File: docker-compose-dev.yml
Comment (shankari) [inline docker-compose-dev.yml:49]: Extraneous whitespace change, please revert the entire file.

File: README.md
Comment (shankari) [inline README.md:68]: you should add an entry for `data_demographics_columns_exclude` now that you have added that config option.

File: pages/data.py
Comment (shankari) [inline pages/data.py:66]: **(future)** Is there a reason that you discard here (in `data.py`) while the other calls to `discard` are in `permissions.py`?
https://github.com/search?q=repo%3Ae-mission%2Fop-admin-dashboard%20discard&type=code

**Review by shankari — CHANGES_REQUESTED** [review]
Almost there.
Just some documentation/formatting cleanup and one clarification request...

Please also indicate "Testing done".
"If it is not tested, it is broken"

File: utils/db_utils.py
Comment (shankari) [reply utils/db_utils.py]: Potential improvement logged here:
https://github.com/e-mission/op-admin-dashboard/issues/51#issuecomment-1709275660

**Comment by shankari** [comment]
@achasmita 

Code looks fine.
Please indicate "Testing done" before I can merge.
"If it is not tested, it is broken"

**Comment by achasmita** [comment]
> @achasmita
> 
> Code looks fine. Please indicate "Testing done" before I can merge. "If it is not tested, it is broken"
Screen Shot 2023-09-12 at 11 15 57 AMYes, testing is done and table looks fine.

**Comment by shankari** [comment]
@achasmita thank you for including the screenshot.
Please do that for all PRs going forward.

However, the testing is not sufficient because it doesn't actually show any of the demographic survey fields.
How do I know if the demographic survey is actually being parsed properly?

Also, please automatically exclude the vast majority of the metadata fields. We don't need the `metadata.key` (which is the same for all entries, or the `metadata.type`, ...

Please updated with more descriptive screenshots that show what you are actually testing.

**Comment by achasmita** [comment]
> @achasmita thank you for including the screenshot. Please do that for all PRs going forward.
> 
> However, the testing is not sufficient because it doesn't actually show any of the demographic survey fields. How do I know if the demographic survey is actually being parsed properly?
> 
> Also, please automatically exclude the vast majority of the metadata fields. We don't need the `metadata.key` (which is the same for all entries, or the `metadata.type`, ...
> 
> Please updated with more descriptive screenshots that show what you are actually testing.

Here is the one after removing metadata.key and including demographic survey fields.Screen Shot 2023-09-12 at 3 32 45 PM

**Comment by shankari** [comment]
Does this only remove `metadata.key`? Please note my comment

> Also, please automatically exclude the vast majority of the metadata fields.

**Comment by achasmita** [comment]
> Does this only remove `metadata.key`? Please note my comment
> 
> > Also, please automatically exclude the vast majority of the metadata fields.

i mean all the columns that starts with metadata.

**Comment by achasmita** [comment]
Here is the screenshot of all the columns it is displaying now:

Screen Shot 2023-09-12 at 3 41 06 PM
Screen Shot 2023-09-12 at 3 42 57 PM
Screen Shot 2023-09-12 at 3 43 22 PM


Screen Shot 2023-09-12 at 3 42 31 PM
Screen Shot 2023-09-12 at 3 43 36 PM

Screen Shot 2023-09-12 at 3 42 06 PM

**Comment by shankari** [comment]
Please push these changes and I can merge

File: utils/db_utils.py
Comment (shankari) [inline utils/db_utils.py:112]: I don't think we should hardcode any of the form ids (such as `aSfdnWs9LE6q8YEF7u9n85`) because I believe they can be different from form to form. But the other changes are fine.

Comment (shankari) [reply utils/db_utils.py:112]: e.g. the form ID for the time use survey, for example, is `time_use_survey_form_v9_1`
https://github.com/sebastianbarry/nrel-openpath-deploy-configs/blob/surveys-info-and-surveys-data/survey-resources/data-json/time-use-survey-form-v9.json

**Review by shankari — APPROVED** [review]
I am going to approve this for now and we can tweak based on feedback.

**Comment by achasmita** [comment]
new screenshots ![Screen Shot 2023-09-12 at 4 26 11 PM
Screen Shot 2023-09-12 at 4 25 50 PM](https://github.com/e-mission/op-admin-dashboard/assets/79387860/6d40745c-c61b-437f-81af-35398fa57eb0)

**Pull request merged**
